src/rust/plugin-registry/src/server.rs

```diff
@@ -125,16 +186,33 @@ impl PluginRegistryService for PluginRegistry {
         todo!()
     }
 
+    #[tracing::instrument(skip(self, request), err)]
     async fn get_generators_for_event_source(
         &self,
-        _request: Request<GetGeneratorsForEventSourceRequestProto>,
+        request: Request<GetGeneratorsForEventSourceRequestProto>,
     ) -> Result<Response<GetGeneratorsForEventSourceResponseProto>, Status> {
-        // Stub implementation, for integration test smoke-test purposes.
-        // Replace with a real implementation soon!
-        let message = _request.into_inner();
-        Ok(Response::new(GetGeneratorsForEventSourceResponseProto {
-            plugin_ids: [message.event_source_id].into_iter().flatten().collect(),
-        }))
+        let request = request.into_inner();
+        let request =
+            GetGeneratorsForEventSourceRequest::try_from(request).expect("Invalid message");
+
+        match self.get_generators_for_event_source(request).await {
```

Comment:
(Discussed in slack)

Reply:
Yeah, this is a bit confusing.

Reply:
I don't disagree but I don't have a good solution. Historically I've prefaced internal methods that share a name with an external method _ to avoid ambiguity.

Another option is to add something like 'impl'.

For now I think it's fine to leave it.